KVM failing test: TestFunctional/parallel/MountCmd

example https://storage.googleapis.com/minikube-builds/logs/10656/dd6687e/KVM_Linux.html

```
=== RUN   TestFunctional/parallel/MountCmd
=== PAUSE TestFunctional/parallel/MountCmd
=== CONT  TestFunctional/parallel/MountCmd
=== CONT  TestFunctional/parallel/MountCmd
fn_mount_cmd_test.go:72: (dbg) daemon: [out/minikube-linux-amd64 mount -p functional-20210302173523-7287 /tmp/mounttest342817430:/mount-9p --alsologtostderr -v=1]
fn_mount_cmd_test.go:106: wrote "test-1614706782178722166" to /tmp/mounttest342817430/created-by-test
fn_mount_cmd_test.go:106: wrote "test-1614706782178722166" to /tmp/mounttest342817430/created-by-test-removed-by-pod
fn_mount_cmd_test.go:106: wrote "test-1614706782178722166" to /tmp/mounttest342817430/test-1614706782178722166
fn_mount_cmd_test.go:114: (dbg) Run:  out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh "findmnt -T /mount-9p | grep 9p"
=== CONT  TestFunctional/parallel/MountCmd
fn_mount_cmd_test.go:114: (dbg) Non-zero exit: out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh "findmnt -T /mount-9p | grep 9p": exit status 1 (440.274675ms)
** stderr ** 
	ssh: Process exited with status 1
** /stderr **
=== CONT  TestFunctional/parallel/MountCmd
fn_mount_cmd_test.go:114: (dbg) Run:  out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh "findmnt -T /mount-9p | grep 9p"
fn_mount_cmd_test.go:114: (dbg) Non-zero exit: out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh "findmnt -T /mount-9p | grep 9p": exit status 1 (291.706818ms)
** stderr ** 
	ssh: Process exited with status 1
** /stderr **
=== CONT  TestFunctional/parallel/MountCmd
fn_mount_cmd_test.go:114: (dbg) Run:  out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh "findmnt -T /mount-9p | grep 9p"
=== CONT  TestFunctional/parallel/MountCmd
fn_mount_cmd_test.go:114: (dbg) Non-zero exit: out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh "findmnt -T /mount-9p | grep 9p": exit status 1 (360.793177ms)
** stderr ** 
	ssh: Process exited with status 1
** /stderr **
=== CONT  TestFunctional/parallel/MountCmd
fn_mount_cmd_test.go:114: (dbg) Run:  out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh "findmnt -T /mount-9p | grep 9p"
=== CONT  TestFunctional/parallel/MountCmd
fn_mount_cmd_test.go:114: (dbg) Non-zero exit: out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh "findmnt -T /mount-9p | grep 9p": exit status 1 (288.610176ms)
** stderr ** 
	ssh: Process exited with status 1
** /stderr **
fn_mount_cmd_test.go:114: (dbg) Run:  out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh "findmnt -T /mount-9p | grep 9p"
fn_mount_cmd_test.go:114: (dbg) Non-zero exit: out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh "findmnt -T /mount-9p | grep 9p": exit status 1 (289.443264ms)
** stderr ** 
	ssh: Process exited with status 1
** /stderr **
fn_mount_cmd_test.go:114: (dbg) Run:  out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh "findmnt -T /mount-9p | grep 9p"
fn_mount_cmd_test.go:114: (dbg) Non-zero exit: out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh "findmnt -T /mount-9p | grep 9p": exit status 1 (353.827389ms)
** stderr ** 
	ssh: Process exited with status 1
** /stderr **
=== CONT  TestFunctional/parallel/MountCmd
fn_mount_cmd_test.go:114: (dbg) Run:  out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh "findmnt -T /mount-9p | grep 9p"
=== CONT  TestFunctional/parallel/MountCmd
fn_mount_cmd_test.go:114: (dbg) Non-zero exit: out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh "findmnt -T /mount-9p | grep 9p": exit status 1 (277.549972ms)
** stderr ** 
	ssh: Process exited with status 1
** /stderr **
=== CONT  TestFunctional/parallel/MountCmd
fn_mount_cmd_test.go:114: (dbg) Run:  out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh "findmnt -T /mount-9p | grep 9p"
=== CONT  TestFunctional/parallel/MountCmd
fn_mount_cmd_test.go:114: (dbg) Non-zero exit: out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh "findmnt -T /mount-9p | grep 9p": exit status 1 (312.044217ms)
** stderr ** 
	ssh: Process exited with status 1
** /stderr **
fn_mount_cmd_test.go:124: /mount-9p did not appear within 17.747581045s: exit status 1
fn_mount_cmd_test.go:79: "TestFunctional/parallel/MountCmd" failed, getting debug info...
fn_mount_cmd_test.go:80: (dbg) Run:  out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh "mount | grep 9p; ls -la /mount-9p; cat /mount-9p/pod-dates"
=== CONT  TestFunctional/parallel/MountCmd
fn_mount_cmd_test.go:80: (dbg) Non-zero exit: out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh "mount | grep 9p; ls -la /mount-9p; cat /mount-9p/pod-dates": exit status 1 (407.087062ms)
-- stdout --
	total 0
	drwxr-xr-x  2 root root  40 Mar  2 17:39 .
	drwxr-xr-x 19 root root 500 Mar  2 17:39 ..
	cat: /mount-9p/pod-dates: No such file or directory
-- /stdout --
** stderr ** 
	ssh: Process exited with status 1
** /stderr **
fn_mount_cmd_test.go:82: debugging command "out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh \"mount | grep 9p; ls -la /mount-9p; cat /mount-9p/pod-dates\"" failed : exit status 1
fn_mount_cmd_test.go:89: (dbg) Run:  out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh "sudo umount -f /mount-9p"
=== CONT  TestFunctional/parallel/MountCmd
fn_mount_cmd_test.go:89: (dbg) Non-zero exit: out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh "sudo umount -f /mount-9p": exit status 1 (393.610895ms)
-- stdout --
	umount: /mount-9p: not mounted.
-- /stdout --
** stderr ** 
	ssh: Process exited with status 32
** /stderr **
fn_mount_cmd_test.go:91: "out/minikube-linux-amd64 -p functional-20210302173523-7287 ssh \"sudo umount -f /mount-9p\"": exit status 1
fn_mount_cmd_test.go:93: (dbg) stopping [out/minikube-linux-amd64 mount -p functional-20210302173523-7287 /tmp/mounttest342817430:/mount-9p --alsologtostderr -v=1] ...
fn_mount_cmd_test.go:93: (dbg) [out/minikube-linux-amd64 mount -p functional-20210302173523-7287 /tmp/mounttest342817430:/mount-9p --alsologtostderr -v=1] stdout:
* Mounting host path /tmp/mounttest342817430 into VM as /mount-9p ...
- Mount type:   
- User ID:      docker
- Group ID:     docker
- Version:      9p2000.L
- Message Size: 262144
- Permissions:  755 (-rwxr-xr-x)
- Options:      map[]
- Bind Address: 192.168.39.29:38969
* Userspace file server: ufs starting
* Userspace file server is shutdown
fn_mount_cmd_test.go:93: (dbg) [out/minikube-linux-amd64 mount -p functional-20210302173523-7287 /tmp/mounttest342817430:/mount-9p --alsologtostderr -v=1] stderr:
I0302 17:39:42.319536    9828 out.go:229] Setting OutFile to fd 1 ...
I0302 17:39:42.319889    9828 out.go:276] TERM=,COLORTERM=, which probably does not support color
I0302 17:39:42.319904    9828 out.go:242] Setting ErrFile to fd 2...
I0302 17:39:42.319911    9828 out.go:276] TERM=,COLORTERM=, which probably does not support color
I0302 17:39:42.320111    9828 root.go:308] Updating PATH: /home/jenkins/minikube-integration/linux-amd64-kvm2--10656-2894-dd6687ecb2d95647810c3fe334f12e92c7227707/.minikube/bin
I0302 17:39:42.320444    9828 mustload.go:66] Loading cluster: functional-20210302173523-7287
I0302 17:39:42.321736    9828 main.go:121] libmachine: Found binary path at /home/jenkins/workspace/KVM_Linux_integration/out/docker-machine-driver-kvm2
I0302 17:39:42.321811    9828 main.go:121] libmachine: Launching plugin server for driver kvm2
I0302 17:39:42.344148    9828 main.go:121] libmachine: Plugin server listening at address 127.0.0.1:35317
I0302 17:39:42.344958    9828 main.go:121] libmachine: () Calling .GetVersion
I0302 17:39:42.345727    9828 main.go:121] libmachine: Using API Version  1
I0302 17:39:42.345756    9828 main.go:121] libmachine: () Calling .SetConfigRaw
I0302 17:39:42.346195    9828 main.go:121] libmachine: () Calling .GetMachineName
I0302 17:39:42.346358    9828 main.go:121] libmachine: (functional-20210302173523-7287) Calling .GetState
I0302 17:39:42.352080    9828 host.go:66] Checking if "functional-20210302173523-7287" exists ...
I0302 17:39:42.352739    9828 main.go:121] libmachine: Found binary path at /home/jenkins/workspace/KVM_Linux_integration/out/docker-machine-driver-kvm2
I0302 17:39:42.352802    9828 main.go:121] libmachine: Launching plugin server for driver kvm2
I0302 17:39:42.370399    9828 main.go:121] libmachine: Plugin server listening at address 127.0.0.1:38909
I0302 17:39:42.371167    9828 main.go:121] libmachine: () Calling .GetVersion
I0302 17:39:42.371896    9828 main.go:121] libmachine: Using API Version  1
I0302 17:39:42.371932    9828 main.go:121] libmachine: () Calling .SetConfigRaw
I0302 17:39:42.372356    9828 main.go:121] libmachine: () Calling .GetMachineName
I0302 17:39:42.372619    9828 main.go:121] libmachine: (functional-20210302173523-7287) Calling .DriverName
I0302 17:39:42.372820    9828 main.go:121] libmachine: (functional-20210302173523-7287) Calling .DriverName
I0302 17:39:42.373049    9828 main.go:121] libmachine: (functional-20210302173523-7287) Calling .GetIP
I0302 17:39:42.391219    9828 main.go:121] libmachine: (functional-20210302173523-7287) Calling .DriverName
I0302 17:39:42.398646    9828 out.go:119] * Mounting host path /tmp/mounttest342817430 into VM as /mount-9p ...
I0302 17:39:42.400992    9828 out.go:119]   - Mount type:   
I0302 17:39:42.403136    9828 out.go:119]   - User ID:      docker
I0302 17:39:42.405268    9828 out.go:119]   - Group ID:     docker
I0302 17:39:42.407363    9828 out.go:119]   - Version:      9p2000.L
I0302 17:39:42.409425    9828 out.go:119]   - Message Size: 262144
I0302 17:39:42.412314    9828 out.go:119]   - Permissions:  755 (-rwxr-xr-x)
I0302 17:39:42.414496    9828 out.go:119]   - Options:      map[]
I0302 17:39:42.416604    9828 out.go:119]   - Bind Address: 192.168.39.29:38969
I0302 17:39:42.416824    9828 ssh_runner.go:149] Run: /bin/bash -c "[ "x$(findmnt -T /mount-9p | grep /mount-9p)" != "x" ] && sudo umount -f /mount-9p || echo "
I0302 17:39:42.416873    9828 main.go:121] libmachine: (functional-20210302173523-7287) Calling .GetSSHHostname
I0302 17:39:42.418806    9828 out.go:119] * Userspace file server: 
I0302 17:39:42.418949    9828 main.go:107] stdlog: ufs.go:27 listen tcp 192.168.39.29:38969: bind: cannot assign requested address
I0302 17:39:42.420937    9828 out.go:119] * Userspace file server is shutdown
I0302 17:39:42.424618    9828 main.go:121] libmachine: (functional-20210302173523-7287) Calling .GetSSHPort
I0302 17:39:42.424808    9828 main.go:121] libmachine: (functional-20210302173523-7287) Calling .GetSSHKeyPath
I0302 17:39:42.424982    9828 main.go:121] libmachine: (functional-20210302173523-7287) Calling .GetSSHUsername
I0302 17:39:42.425149    9828 sshutil.go:53] new ssh client: &{IP:192.168.39.29 Port:22 SSHKeyPath:/home/jenkins/minikube-integration/linux-amd64-kvm2--10656-2894-dd6687ecb2d95647810c3fe334f12e92c7227707/.minikube/machines/functional-20210302173523-7287/id_rsa Username:docker}
I0302 17:39:42.615584    9828 mount.go:147] unmount for /mount-9p ran successfully
I0302 17:39:42.615626    9828 ssh_runner.go:149] Run: /bin/bash -c "sudo mkdir -m 755 -p /mount-9p"
I0302 17:39:42.631783    9828 ssh_runner.go:149] Run: /bin/bash -c "sudo mount -t 9p -o dfltgid=$(grep ^docker: /etc/group | cut -d: -f3),dfltuid=$(id -u docker),msize=262144,port=38969,trans=tcp,version=9p2000.L 192.168.39.29 /mount-9p"
I0302 17:39:42.692337    9828 out.go:119] 
W0302 17:39:42.692588    9828 out.go:181] X Exiting due to GUEST_MOUNT: mount with cmd /bin/bash -c "sudo mount -t 9p -o dfltgid=$(grep ^docker: /etc/group | cut -d: -f3),dfltuid=$(id -u docker),msize=262144,port=38969,trans=tcp,version=9p2000.L 192.168.39.29 /mount-9p" : /bin/bash -c "sudo mount -t 9p -o dfltgid=$(grep ^docker: /etc/group | cut -d: -f3),dfltuid=$(id -u docker),msize=262144,port=38969,trans=tcp,version=9p2000.L 192.168.39.29 /mount-9p": Process exited with status 32
stdout:
stderr:
mount: /mount-9p: mount(2) system call failed: Connection refused.
X Exiting due to GUEST_MOUNT: mount with cmd /bin/bash -c "sudo mount -t 9p -o dfltgid=$(grep ^docker: /etc/group | cut -d: -f3),dfltuid=$(id -u docker),msize=262144,port=38969,trans=tcp,version=9p2000.L 192.168.39.29 /mount-9p" : /bin/bash -c "sudo mount -t 9p -o dfltgid=$(grep ^docker: /etc/group | cut -d: -f3),dfltuid=$(id -u docker),msize=262144,port=38969,trans=tcp,version=9p2000.L 192.168.39.29 /mount-9p": Process exited with status 32
stdout:
stderr:
mount: /mount-9p: mount(2) system call failed: Connection refused.
W0302 17:39:42.693024    9828 out.go:181] * 
* 
W0302 17:39:42.693070    9828 out.go:181] * If the above advice does not help, please let us know: 
* If the above advice does not help, please let us know: 
W0302 17:39:42.693118    9828 out.go:181]   - https://github.com/kubernetes/minikube/issues/new/choose
- https://github.com/kubernetes/minikube/issues/new/choose
I0302 17:39:42.697927    9828 out.go:119] 
--- FAIL: TestFunctional/parallel/MountCmd (18.66s)
```

--- a/pkg/minikube/cluster/ip.go
+++ b/pkg/minikube/cluster/ip.go
@@ -47,11 +47,17 @@ func HostIP(host *host.Host, clusterName string) (net.IP, error) {
 		}
 		return net.ParseIP(ip), nil
 	case driver.KVM2:
-		ip, err := host.Driver.GetIP()
+		// `host.Driver.GetIP` returns dhcp lease info for a given network(=`virsh net-dhcp-leases minikube-net`)
+		vmIPString, err := host.Driver.GetIP()
 		if err != nil {
 			return []byte{}, errors.Wrap(err, "Error getting VM/Host IP address")
 		}
-		return net.ParseIP(ip), nil
+		vmIP := net.ParseIP(vmIPString).To4()
+		if vmIP == nil {
+			// We need the network ip address for minikube-net. It's the start address of the returned subnet.
+			return []byte{}, errors.Wrap(err, "Error converting VM/Host IP address to IPv4 address")
+		}
+		return net.IPv4(vmIP[0], vmIP[1], vmIP[2], byte(1)), nil
 	case driver.HyperV:
 		v := reflect.ValueOf(host.Driver).Elem()
 		var hypervVirtualSwitch string

